sklearn/linear_model/coordinate_descent.py

```diff
@@ -2033,3 +2034,168 @@ def __init__(self, eps=1e-3, n_alphas=100, alphas=None, fit_intercept=True,
+        to false, no intercept will be used in calculations
+        (e.g. data is expected to be already centered).
+    normalize : boolean, optional, default False
+        If ``True``, the regressors X will be normalized before regression.
+    copy_X : boolean, optional, default True
+        If ``True``, X will be copied; else, it may be overwritten.
+    precompute : True | False | 'auto' | array-like
+        Whether to use a precomputed Gram matrix to speed up
+        calculations. If set to ``'auto'`` let us decide. The Gram
+        matrix can also be passed as argument. For sparse input
+        this option is always ``True`` to preserve sparsity.
+        WARNING : The ``'auto'`` option is deprecated and will
+        be removed in 0.18.
+    max_iter : int, optional
+        The maximum number of iterations
+    tol : float, optional
+        The tolerance for the optimization: if the updates are
+        smaller than ``tol``, the optimization code checks the
+        dual gap for optimality and continues until it is smaller
+        than ``tol``.
+    positive : bool, optional
+        When set to ``True``, forces the coefficients to be positive.
+    selection : str, default 'cyclic'
+        If set to 'random', a random coefficient is updated every iteration
+        rather than looping over features sequentially by default. This
+        (setting to 'random') often leads to significantly faster convergence
+        especially when tol is higher than 1e-4.
+    random_state : int, RandomState instance, or None (default)
+        The seed of the pseudo random number generator that selects
+        a random feature to update. Useful only when selection is set to
+        'random'.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) | (n_targets, n_features)
+        parameter vector (w in the cost function formula)
+
+    sparse_coef_ : scipy.sparse matrix, shape (n_features, 1) | \
+            (n_targets, n_features)
+        ``sparse_coef_`` is a readonly property derived from ``coef_``
+
+    intercept_ : float | array, shape (n_targets,)
+        independent term in decision function.
+
+    n_iter_ : int | array-like, shape (n_targets,)
+        number of iterations run by the coordinate descent solver to reach
+        the specified tolerance.
+
+    See also
+    --------
+    Lasso
+
+    Notes
+    -----
+    The algorithm used to fit the model is coordinate descent.
+
+    Zou, H., & Li, R.
+    One-step sparse estimates in nonconcave penalized likelihood models.
+    The Annals of Statistics Ann. Statist., 1509-1533.
+
+    Buhlmann, P., & Meier, L.
+    Discussion: One-step sparse estimates in nonconcave penalized likelihood
+    models.
+    The Annals of Statistics Ann. Statist., 1534-1541.
+
+    Zou, Hui.
+    The Adaptive Lasso and Its Oracle Properties
+    Journal of the American Statistical Association
+    """
+    def __init__(self, n_lasso_iterations=2, gamma=1, alpha=1.0,
```

Comment:
>
> Well I was thinking that the adaptive lasso corresponds to 2 steps, but
> this could be increased to more (5?).
>
you should iterate until the cost function stop decreasing up to a given
tolerance. My experience with neuroscience data is that 5 to 10 iterations
is enough.

> About eps: 0.001 was the value they used in another article (added ref),
> that's why I figured it was a sensible default.
>
hum ok...